File: LeetCode/1.js
// Two Sum
const twoSum = function (nums, target) {
	for (let i = 0; i < nums.length; ++i) {
		const need = target - nums[i];
		for (let j = 0; j < nums.length; ++j) {
			if (nums[j] === need && j !== i) {
				return [i, j];
			}
		}
	}
	return -1;
};

File: LeetCode/src/1.cpp
// Two Sum
#include <map>
#include <vector>
using namespace std;

class Solution {
public:
  vector<int> twoSum(vector<int> &nums, int target) {
    map<int, int> hash;
    for (int i = 0; i < nums.size(); ++i) {
      int need = target - nums[i];
      if (hash.find(need) != hash.end()) {
        return {hash[need], i};
      }
      hash[nums[i]] = i;
    }
    return {-1};
  }
};